### PR TITLE
Minor improvements of lib/msf/core/payload/php.rb

### PR DESCRIPTION
**Pull request by jvoisin**
- Golf a condition
- Use backticks instead of `passthru`, as they're equivalent
- Use the `shuffle` method instead of the weird `.sort_by` construct

### CODE REVIEW

File: lib/msf/core/payload/php.rb [hunk 1]
```diff
@@ -82,7 +82,7 @@ def php_system_block(options = {})
     passthru = "
       if(#{is_callable}('passthru')&&!#{in_array}('passthru',#{dis})){
         ob_start();
-        passthru(#{cmd});
+        `#{cmd}`;
```
Comment (sjanusz-r7) [inline lib/msf/core/payload/php.rb:85]: Could we get some clarification/reasoning behind this change?

This now relies on `shell_exec`. However, if we get to this point, we know that `shell_exec` is disabled, based on the lines above: https://github.com/rapid7/metasploit-framework/pull/19445/files#diff-964542ddc7e6e035ff04c32c78aae505d165933654eee96f05002f955d538d65R78-R81

Comment (jvoisin) [reply lib/msf/core/payload/php.rb:85]: The reason is me misremembering what ``` was synonym of, sorry :/

**Review by cgranleese-r7 — APPROVED** [review]

**Comment by cgranleese-r7** [comment]
## Release Notes
Makes minor improvements of `lib/msf/core/payload/php.rb`.

File: lib/msf/core/payload/php.rb [hunk 2]
```diff
@@ -134,7 +134,8 @@ def php_system_block(options = {})
       }
     "
 
-    exec_methods = [passthru, shell_exec, system, exec, proc_open, popen].sort_by { rand }
+    exec_methods = [passthru, shell_exec, system, exec, proc_open, popen];
+    shuffle(exec_methods);
```
Comment (Chocapikk) [inline lib/msf/core/payload/php.rb:138]: Hey @jvoisin, there's an issue with this

Here is the fix: 

```suggestion
    exec_methods = %w(passthru shell_exec system exec proc_open popen)
    exec_methods.shuffle!
```

I was working on a new module and had this error:

`Exploit failed: NoMethodError undefined method shuffle`

Comment (jvoisin) [reply lib/msf/core/payload/php.rb:138]: https://github.com/rapid7/metasploit-framework/pull/19491 !

**Pull request merged**